src/dns_parser/name.rs

```diff
@@ -61,6 +61,9 @@ impl<'a> Name<'a> {
                 ));
             } else if byte & 0b1100_0000 == 0 {
                 let end = pos + byte as usize + 1;
+                if data.len() <= end {
```

Comment:
nit: I think it would be slightly clearer to write `if end >= data.len()`

Reply:
That's how I feel too and I had it initially but then changed it to this to match the majority of the other comparisons in this function.

I'll change it back.  When upstreaming will just go with whatever is preferred.